app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -197,6 +197,28 @@ def vm_compute_attributes_for(uuid)
       vm_attrs
     end
 
+    def normalize_vm_attrs(vm_attrs)
+      vm_attrs['image_name'] = self.images.where(:uuid => vm_attrs['image_id']).first.try(:name)
+      vm_attrs['volumes_attributes'] ||= {}
+      vm_attrs['volumes_attributes'].each do |key, vol|
+        vol['capacity'] = memory_gb_to_bytes(vol['capacity']).to_s
+        vol['allocation'] = memory_gb_to_bytes(vol['allocation']).to_s
+        vol['pool'] = vol.delete('pool_name')
+      end
+      vm_attrs['interfaces_attributes'] = vm_attrs.delete('nics_attributes') || {}
+      vm_attrs['interfaces_attributes'].each do |key, nic|
+        # Remove the value for 'bridge' when the type is 'network' and vice versa
+        # The value isn't used anyway. It gets stored only when user first selects the bridge
+        # and then switches the type to 'network'.
```

Comment:
would this be consistent when we use the same format for creation? how would we differentiate between networks and bridges?